parsec/interfaces/dtd/parsec_dtd_data_flush.c

```diff
@@ -326,15 +322,15 @@ parsec_dtd_insert_flush_task_pair(parsec_taskpool_t *tp, parsec_dtd_tile_t *tile
         if(last_writer.task->rank != tile->rank) {
```

Comment:
Actually every process sees that task. It's not inserted on just one process but only if the last writer of the  tile is not the same as the owner.

Reply:
Yes, I misread the code and @therault pointed me on the right direction. Why is the sending task missing one ref count ?

Reply:
So here is what I found while debugging:

The flush task is created (https://github.com/ICLDisco/parsec/blob/c4cc5b01e46552cef018f90f05bd601ec6d0b350/parsec/interfaces/dtd/parsec_dtd_data_flush.c#L266) "manually" (i.e., not using `parsec_dtd_insert_task`) and then passed to `parsec_insert_dtd_flush_task` where it is retained (https://github.com/ICLDisco/parsec/blob/master/parsec/interfaces/dtd/parsec_dtd_data_flush.c#L134) and released again (https://github.com/ICLDisco/parsec/blob/master/parsec/interfaces/dtd/parsec_dtd_data_flush.c#L239) at the end of the function. That's according to plan, as far as I can see. When the second task of the pair is created, that first task is retained when it's new descendant is set (https://github.com/ICLDisco/parsec/blob/master/parsec/interfaces/dtd/parsec_dtd_data_flush.c#L185). This is good, because eventually this reference will be released once the dependency is released. However, we also release that first task right away because it's not the last writer anymore (https://github.com/ICLDisco/parsec/blob/master/parsec/interfaces/dtd/parsec_dtd_data_flush.c#L225), which is correct (we don't care about tasks anymore if they are not the last writer anymore). It's important though to keep that one reference that will be released once the dependency is released. That's the reference that is added if `is_first_flush_task` is `true`.

Maybe the original sin is that the code doesn't use `parsec_dtd_insert_task` (which hopefully would do just the right thing) but I cannot properly judge that. And I don't know why it wasn't done that way in the first place. My sense is that my solution is a hack, piled on top of more hacks. The manual ref-counting (e.g., https://github.com/ICLDisco/parsec/blob/c4cc5b01e46552cef018f90f05bd601ec6d0b350/parsec/interfaces/dtd/parsec_dtd_data_flush.c#L275) makes reasoning about the life-time hard. I'd prefer doing this properly but I'm afraid that pulling one string will unravel all others.

Reply:
I believe this conversation (tile owner rank != tile last user rank) is resolved. Re-open it if I'm wrong?